cpp/ql/lib/semmle/code/cpp/ir/implementation/aliased_ssa/internal/SSAConstruction.qll

```diff
@@ -365,6 +389,18 @@ private module Cached {
     result = getNewPhiOperandDefinitionFromOldSsa(instr, newPredecessorBlock, overlap)
   }
 
+  private ChiInstruction getChiAfterInitializeGroup(int i, IRFunction func) {
+    result =
+      rank[i + 1](VariableGroup vg, InitializeGroupInstruction initGroup, ChiInstruction chi, int r |
+        initGroup.getEnclosingIRFunction() = func and
+        chi = getChi(initGroup) and
+        initGroup = initializeGroup(vg) and
+        r = vg.getInitializationOrder()
+      |
+        chi order by r
+      )
+  }
```

Comment:
So we're numbering the Chi instructions separately from the initialize instructions.  Am I right in understanding that this numbering is only used to pass aliased memory from one Chi to the next, whereas the instructions themselves are sequenced according to the VariableGroup ordering (with Chi's interleaved)?

Reply:
Yep, that's exactly right. We use `VariableGroup.getInitializationOrder` to order the initialize instructions, and then we interleave those with `Chi` instructions if necessary. The order is only important because we need to pass aliased memory from one Chi to the next (whenever an allocation in a `VariableGroup` escapes).